apps/src/templates/teacherDashboard/SectionActionDropdown.jsx

```diff
@@ -0,0 +1,271 @@
+  },
+  heading: {
+    borderTopWidth: 0,
+    borderBottomWidth: 1,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.default_text,
+    paddingBottom: 20,
+    marginBottom: 30,
+  },
+};
+
+class SectionActionDropdown extends Component {
+  static propTypes = {
+    handleEdit: PropTypes.func,
+    sectionData: sortableSectionShape.isRequired,
+
+    //Provided by redux
+    removeSection: PropTypes.func.isRequired,
+    toggleSectionHidden: PropTypes.func.isRequired,
+    sectionCode: PropTypes.string,
+    sectionName: PropTypes.string,
+    updateRoster: PropTypes.func.isRequired,
+  };
+
+  state = {
+    deleting: false,
+    open: false,
+    canOpen: true,
+    menuTop: 0, // location of dropdown menu
+    menuLeft: 0,
+    currWindowWidth: window.innerWidth, // used to calculate location of menu on resize
+  };
+
+  // Menu open
+  open = () => {
+    this.updateMenuLocation();
+    window.addEventListener("resize", this.updateMenuLocation);
+    this.setState({open: true, canOpen: false});
+  }
+
+  // Menu closed
+  close = () => {
+    window.removeEventListener("resize", this.updateMenuLocation);
+    this.setState({open: false});
+  }
+
+  // Menu closed
+  beforeClose = (_, resetPortalState) => {
+    resetPortalState();
+    this.setState({open: false});
+    window.setTimeout(() => this.setState({canOpen: true}), 0);
+  };
+
+  onConfirmDelete = () => {
+      const {removeSection } = this.props;
+      const section = this.props.sectionData;
+      $.ajax({
+          url: `/v2/sections/${section.id}`,
+          method: 'DELETE',
+      }).done(() => {
+          removeSection(section.id);
+      }).fail((jqXhr, status) => {
+          // We may want to handle this more cleanly in the future, but for now this
+          // matches the experience we got in angular
+          alert(i18n.unexpectedError());
+          console.error(status);
+      });
+  };
+
+  onClickEdit = () => {
+      this.props.handleEdit(this.props.sectionData.id);
+  };
+
+  onClickHideShow = () => {
+      this.props.toggleSectionHidden(this.props.sectionData.id);
+  };
+
+  onClickSync = () => {
+    // Section code is the course ID, without the G- or C- prefix.
+    const courseId = this.props.sectionCode.replace(/^[GC]-/, '');
+    this.props.updateRoster(courseId, this.props.sectionName);
+  };
+
+  onRequestDelete = () => {
+    this.setState({deleting: true});
+  };
+
+  onCancelDelete = () => {
+    this.setState({deleting: false});
+  }
+
+  updateMenuLocation = () => {
+    const rect = this.icon.getBoundingClientRect();
+    const windowWidth = window.innerWidth;
+    if (windowWidth > 970) { // Accounts for resizing when page is not scrollable
+      this.setState({menuTop: rect.bottom + window.pageYOffset});
+      this.setState({menuLeft: rect.left - rect.width - (windowWidth - this.state.currWindowWidth)/2});
+      this.setState({currWindowWidth : window.innerWidth});
```

Comment:
Prefer to make only one `setState()` call from any given path through a function, since it triggers React behavior.
```js
this.setState({
  menuTop: rect.bottom + window.pageYOffset,
  menuLeft: rect.left - rect.width - (windowWidth - this.state.currWindowWidth)/2,
  currWindowWidth : window.innerWidth,
});
```